CONTRIBUTING.md

```diff
@@ -0,0 +1,147 @@
+The [OOT Decompilation Project](https://github.com/zeldaret/oot) is farther along than this project, so it can also be a great resource.
+
+This project only uses *publicly available code*.
+Anyone who wishes to contribute to the OOT or MM projects **must not have accessed leaked source code at any point in time** for Nintendo 64 SDK, iQue player SDK, libultra, Ocarina of Time, Majora's Mask, Animal Crossing/Animal Forest, or any other game that shares the same game engine or significant portions of code to a Zelda 64 game or any other console similar to the Nintendo 64.
+
+### Environment Setup
+
+Get started by following the [installation guide in the readme](https://github.com/zeldaret/mm/blob/master/README.md#installation).
+When successful, you should be able to build a matching ROM before you start making any changes.
+
+### First Contribution
+
+Usually, the best place to get started is to decompile an actor overlay. 
+An *actor* is any thing in the game that moves or performs actions or interactions: Link is an actor, enemies are actors, NPCs are actors, etc.
+
+You should [join the Discord](https://discord.zelda64.dev/) to say hello and get suggestions on where to start on the `#mm-decomp` channel.
+
+We track who is working on what on the [Trello board](https://trello.com/b/ruxw9n6m/majoras-mask-decompilation).
+After joining the Discord, ask to be added to the Trello board so you can claim your actor and avoid duplicate work.
+
+Style Guide & Conventions
+-------------------------
+
+Most of the C formatting style is enforced by the `format.sh` script, which is based on `clang-format`.
+Running `./format.sh` will apply our standard style to all `.c` files in the repository.
+
+There are some conventions that cannot be automatically enforced.
+
+### Naming Scheme
+
+The following overlays are good examples of our naming conventions:
+
+- [`ovl_Dm_Ravine`](src/overlays/actors/ovl_Dm_Ravine/z_dm_ravine.c)
+- [`ovl_En_Jc_Mato`](src/overlays/actors/ovl_En_Jc_Mato/z_en_jc_mato.c)
+
+These files demonstrate the following:
+
+- Word order in names are from least-to-most specific (`DM_RAVINE_STATE_ACTIVE`, not `DM_RAVINE_ACTIVE_STATE`)
+- Functions, structs, unions, enums, and typedefs are `TitleCase` (`DmRavine`)
+    - "Methods" for objects separate the object from the verb with an underscore (`DmRavine_Init`)
+- Variable names are `camelCase` (`actionFunc`)
+    - Global variables start with `g` (`gSaveContext`)
+    - Static global variables start with `s` (`sSphereInit`)
+- Macros and enum constants are `SCREAMING_SNAKE_CASE` (`DM_RAVINE_STATE_ACTIVE`)
+- Auxiliary structs, enums, and constants are defined in the header file
+- Trailing commas in array and struct definitions (see `EnJcMatoDamageTable`)
+- Constants converted to whichever looks best in context: hexadecimal, decimal, float, or boolean
+    - Rotation angles should always be in hexadecimal
+- Define macros for bitwise access to `actor->params`
+
+### `NON_MATCHING`
```

Comment:
We should at least mention the existence of `NON_EQUIVALENT` too.

Reply:
What's the difference? 

I think explaining `NON_MATCHING` is probably good enough. I think the key point I wanted to convey here is that "if you get a function really close but you can't finish it and you're going to give up, consider `NON_MATCHING` instead of throwing it out."